utils/guardrails/guard.py

```diff
@@ -0,0 +1,204 @@
+                params["sambastudio_api_key"] = sambastudio_api_key
+            self.llm = self.set_llm("sambastudio", params)
+        elif api == "sambaverse":
+            params = {}
+            if sambaverse_base_url:
+                params["sambaverse_base_url"] = sambaverse_base_url
+            if sambaverse_api_key:
+                params["sambaverse_api_key"] = sambaverse_api_key
+            self.llm = self.set_llm("sambaverse", params)
+    
+    def load_guardrails(self, path: str):
+        """
+        Load enabled guardrails from a YAML file and return them as a dictionary and a formatted string.
+
+        Parameters:
+        - path (str): The path to the YAML file containing the guardrails.
+
+        Returns:
+        - guardrails (dict): A dictionary of guardrails, where the keys are the guardrail IDs and the values are dictionaries containing the guardrail name and details.
+        - guardrails_str (str): A formatted string containing the names and descriptions of the enabled guardrails.
+        """
+        with open(path, 'r') as yaml_file:
+            guardrails = yaml.safe_load(yaml_file)
+        enabled_guardrails = {k: v for k, v in guardrails.items() if v.get('enabled')}
+        guardrails_list = [f"{k}: {v.get('name')}\n{v.get('description')}" for k, v in enabled_guardrails.items()]
+        guardrails_str = "\n".join(guardrails_list)
+        return guardrails, guardrails_str
+    
+    def set_llm(self, api: str, params: Optional[Dict]):
+        """
+        Set the LLM based on the provided API and parameters.
+
+        Parameters:
+        - api (str): The LLM API to use, either 'sambaverse' or 'sambastudio'.
+        - params (Dict, optional): Additional parameters for the LLM.
+
+        Returns:
+        - LLM: An instance of the specified LLM.
+
+        Raises:
+        - ValueError: If an invalid API is provided.
+        """
+        if api == "sambastudio":
+            llm = SambaStudio(
+                **params,
+                model_kwargs={
+                "select_expert": "Meta-Llama-Guard-2-8B",
+                "process_prompt": False,
+                "do_sample": False,
+                "max_tokens_to_generate": 1024,
+                "temperature": 0.1,
+                },
+            )
+        elif api == "sambaverse":
+            llm = Sambaverse(
+                **params,
+                sambaverse_model_name="Meta/Meta-Llama-Guard-2-8B",
+                model_kwargs={
+                    "select_expert": "Meta-Llama-Guard-2-8B",
+                    "process_prompt": True,
+                    "do_sample": False,
+                    "max_tokens_to_generate": 1024,
+                    "temperature": 0.1,
+                },
+            )
+        else:
+            raise ValueError(f"Invalid LLM API: {api}, only'sambastudio' and'sambaverse' are supported.")
+        return llm
+
+    def evaluate(
+        self,
+        input: Union[List[Dict], str],
```

Comment:
We could rename this parameter otherwise than the too generic `input`.
Examples: `input_prompt` or `input_query`.

Reply:
renamed